### PR TITLE
Ionic Scaffold

### PR DESCRIPTION
**Pull request by blake-futchi**
# [PT Story](https://www.pivotaltracker.com/story/show/172057097)
## User Story
```
As a development team,
In order to be able to start adding features,
We would like to have a basic mobile application structure.
```
## Changes proposed in this pull request:
* Sets up base Ionic Scaffold
## What I have learned working on this feature:
* Setting up Ionic Scaffold
## Packages/ added
    -"@capacitor/core": "1.5.2",
    -"@ionic/react": "^5.0.0",
    - "@ionic/react-router": "^5.0.0",
    - "@testing-library/jest-dom": "^4.2.4",
    -"@testing-library/react": "^9.4.0",
    -"@testing-library/user-event": "^8.0.3",
    -"@types/jest": "^24.0.25",
    -"@types/node": "^12.12.24",
    -"@types/react": "^16.9.17",
    -"@types/react-dom": "^16.9.4",
    -"@types/react-router": "^5.1.4",
    -"@types/react-router-dom": "^5.1.3",
    -"ionicons": "^5.0.0",
    -"react": "^16.12.0",
    -"react-dom": "^16.12.0",
    -"react-router": "^5.1.2",
    -"react-router-dom": "^5.1.2",
    -"react-scripts": "3.4.0",
    -"typescript": "3.7.4"

## Screenshots (Only if client side story)
Screen Shot 2020-03-31 at 2 56 46 PM

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import '@ionic/react/css/padding.css';
+import '@ionic/react/css/float-elements.css';
+import '@ionic/react/css/text-alignment.css';
+import '@ionic/react/css/text-transformation.css';
+import '@ionic/react/css/flex-utils.css';
+import '@ionic/react/css/display.css';
+
+/* Theme variables */
+// import './theme/variables.css';
+
```
Comment (kaylawoodbury) [inline src/App.jsx:27]: Do we want to leave these comments in?

File: src/components/ArticleList.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { IonContent } from '@ionic/react';
+
+const ArticleList = () => {
+  return (
+    <IonContent align="center">
+      This is where the articles go
+    </IonContent>
+  )
+}
+
```
Comment (kaylawoodbury) [inline src/components/ArticleList.jsx:11]: Is this just initial testing?

File: src/components/Menu.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  return (
+    <IonSegment>
+      <IonSegmentButton>All News</IonSegmentButton>
+      <IonSegmentButton>Latest News</IonSegmentButton>
+      <IonSegmentButton>Culture</IonSegmentButton>
+      <IonSegmentButton>Tech</IonSegmentButton>
+      <IonSegmentButton>Food</IonSegmentButton>
+      <IonSegmentButton>Sports</IonSegmentButton>
+    </IonSegment>
+
```
Comment (kaylawoodbury) [inline src/components/Menu.jsx]: Format here to remove extra lines

File: src/index.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+import * as serviceWorker from './serviceWorker';
+
+ReactDOM.render(<App />, document.getElementById('root'));
+
+// If you want your app to work offline and load faster, you can change
+// unregister() to register() below. Note this comes with some pitfalls.
+// Learn more about service workers: https://bit.ly/CRA-PWA
```
Comment (kaylawoodbury) [inline src/index.jsx]: Do we want to leave this comment in?

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (kaylawoodbury) [inline .gitignore]: was there anything else we need to add here? Seems very empty compared to the react front ends so wasn't sure if the autogenerated gitignore was empty to begin with

**Review by kaylawoodbury — COMMENTED** [review]
Overall it looks fine to me, not familiar with Ionic so not sure about the code itself. Left some comments about formatting, comments and a few questions I was unsure about.

**Comment by pierre-1** [comment]
Looks good for an initial scaffold. Any files we need to add to a .gitignore?

File: src/App.jsx (hunk 1 above)
Comment (blake-futchi) [reply src/App.jsx:27]: yes just in case we decide to do the dark theme

File: src/components/ArticleList.jsx (hunk 2 above)
Comment (blake-futchi) [reply src/components/ArticleList.jsx:11]: yes, we were making sure components could render properly but this is just a scaffold

File: .gitignore (hunk 5 above)
Comment (blake-futchi) [reply .gitignore]: updating shortly

File: src/index.jsx (hunk 4 above)
Comment (blake-futchi) [reply src/index.jsx]: deleted

File: src/App.jsx (hunk 1 above)
Comment (kaylawoodbury) [reply src/App.jsx:27]: okie doke

File: src/components/ArticleList.jsx (hunk 2 above)
Comment (kaylawoodbury) [reply src/components/ArticleList.jsx:11]: cool

File: src/components/Menu.jsx (hunk 3 above)
Comment (kaylawoodbury) [reply src/components/Menu.jsx]: Looks good

File: src/index.jsx (hunk 4 above)
Comment (kaylawoodbury) [reply src/index.jsx]: thanks

File: .gitignore (hunk 5 above)
Comment (kaylawoodbury) [reply .gitignore]: ah cool, yeah the updated .gitignore looks more like what I expected

**Pull request merged**